aipl/ops/python.py

```diff
@@ -1,19 +1,42 @@
-from aipl import defop
+import sys
+import traceback
+
+from aipl import defop, InnerPythonException
+
+
+def inner_exec(obj, *args, **kwargs):
+    try:
+        return exec(obj, *args, **kwargs)
+    except Exception as e:
+        exc_type, exc_value, exc_traceback = sys.exc_info()
+        tb = traceback.extract_tb(exc_traceback)
+        raise InnerPythonException(exc_value, tb[1:], obj)
+
+
+def inner_eval(obj, *args, **kwargs):
+    try:
+        return eval(obj, *args, **kwargs)
+    except Exception as e:
+        exc_type, exc_value, exc_traceback = sys.exc_info()
+        tb = traceback.extract_tb(exc_traceback)
+        raise InnerPythonException(exc_value, tb[1:], obj)
```

Comment:
As the except blocks are the same here would it be better to do:

```python
def with_inner_python_exception(function, args, kwargs):
   try:
      return function(*args, *kwargs)
   except Exception as e:
      ...
```

Alternatively move the `exc_info` and `extract_tb` into the InnerPythonException constructor (or a `from_exception`) and inline the try-except?

Reply:
I'm inclined to leave it for now, esp since they're in the same place. 
 https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)  We're at the "wince" stage.

Reply:
Fair enough. I tend to treat the "three" as a guideline and do it earlier if the abstraction is obvious (or later if it's non-obvious.)